controlplane/controllers/ck8scontrolplane_controller.go

```diff
@@ -309,21 +309,18 @@ func (r *CK8sControlPlaneReconciler) SetupWithManager(ctx context.Context, mgr c
 
 // ClusterToCK8sControlPlane is a handler.ToRequestsFunc to be used to enqueue requests for reconciliation
 // for CK8sControlPlane based on updates to a Cluster.
-func (r *CK8sControlPlaneReconciler) ClusterToCK8sControlPlane(ctx context.Context, log *logr.Logger) handler.MapFunc {
-	return func(ctx context.Context, o client.Object) []ctrl.Request {
-		c, ok := o.(*clusterv1.Cluster)
-		if !ok {
-			r.Log.Error(nil, fmt.Sprintf("Expected a Cluster but got a %T", o))
-			return nil
-		}
-
-		controlPlaneRef := c.Spec.ControlPlaneRef
-		if controlPlaneRef != nil && controlPlaneRef.Kind == "CK8sControlPlane" {
-			return []ctrl.Request{{NamespacedName: client.ObjectKey{Namespace: controlPlaneRef.Namespace, Name: controlPlaneRef.Name}}}
-		}
+func (r *CK8sControlPlaneReconciler) ClusterToCK8sControlPlane(_ context.Context, o client.Object) []ctrl.Request {
+	c, ok := o.(*clusterv1.Cluster)
+	if !ok {
+		panic(fmt.Sprintf("Expected a Cluster but got a %T", o))
```

Comment:
Any particular reason why we choose to panic here instead of returning an error? I assume this is something we should never have normally?

Reply:
We're passing this function to ``EnqueueRequestsFromMapFunc`` and it doesn't allow returning an error. Indeed, we shouldn't get here normally, the tests should catch it if we pass anything other than a ``Cluster`` object.